qiskit_experiments/curve_analysis/curve_data.py

```diff
@@ -137,3 +139,205 @@ class ParameterRepr:
 
     # Unit
     unit: Optional[str] = None
+
+
+class OptionsDict(dict):
```

Comment:
Maybe this is a bit overkill but it seems like `OptionsDict` could be abstract with `format` as an abstract method.

Reply:
I think `OptionsDict` can be used as a dict extension without formatting. If we make it abstract, we cannot create instance of it.